syntax = "proto3";

package evercoinx.analyzer;

import "google/protobuf/timestamp.proto";

service Analyzer {
    rpc SendFrame(FrameRequest) returns (EmptyResponse) {}
}

message FrameRequest {
    int32 window_index = 1;
    int32 frame_index = 2;
    int32 hand_number = 3;
    google.protobuf.Timestamp hand_time = 4;
    Money total_pot = 5;
    int32 dealer_position = 6;
    repeated Seat seats = 7;
    repeated string board = 8;
}

message EmptyResponse {}

message Money {
    enum Currency {
        UNSET = 0;
        EURO = 1;
        DOLLAR = 2;
    }

    Currency currency = 1;
    double amount = 2;
}

message Seat {
    enum Action {
        UNSET = 0;
        BET = 1;
        RAISE = 2;
        CALL = 3;
        FOLD = 4;
        CHECK = 5;
        ALL_IN = 6;
        SITTING_IN = 7;
        WAITING_FOR_BB = 8;
    }

    sint32 number = 1;
    Action action = 2;
    Money stake = 3;
    Money balance = 4;
    bool playing = 5;
}